react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange() {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit() {
+    event.preventDefault();
+    console.log('state: ', this.state);
+  }
+
+  render() {
+    return (
+    <form>
+      <input type="email" onChange={this.handleChange}></input>
```

Comment:
You need to control the input by setting its value with state, also `input` should be a self closing tag:

```jsx
<input type="email" onChange={this.handleChange} value={this.state.email} />
```